### ISSUE TITLE
MOVEMAIL - The parameter folder to function getFolder is required but was not passed in.

### ISSUE THREAD

**Issue by HenryGelstor** [Tropicalista/lucee-cfimap#20]
This tag works great for me except I've got a problem moving messages.
<cfimap action="MOVEMAIL" connection="Conn" folder="Inbox" messagenumber="1"  newfolder="Test"`>

'The parameter folder to function getFolder is required but was not passed in.
The error occurred on line 135 in file WEB-INF/lucee/library/tag/imap/imap.cfc.  '

Around Line 135 looks like this;
public function moveMail( required connection, required string newFolder, string messageNumber, string uid, string folder ){
		var objFolder = getFolder( arguments.folder );
		var objNewFolder = getFolder( arguments.newFolder );

Geeze looks OK to me.  So not really knowing what I'm doing, I saw other calls like this elsewhere:
  		var objFolder = getFolder( arguments.connection, arguments.folder );
So I added arguments.connection, restarted Lucee no difference.

I also tried changing line 41 of the other imap.cfc...	MoveMail: ['newFolder'], to MoveMail: ['newFolder', 'folder'], in the 	this.metadata.requiredAttributesPerAction section. 

And many other futile things like using uid, checking the folder name I'm using works on other operations, looked at the case "movemail": section

I am completely stumped- looks like it should work, doesn't it?

**Comment by Tropicalista** [comment]
What version of Lucee are you using?

**Comment by HenryGelstor** [comment]
*(4.5.2.018)*

**Comment by HenryGelstor** [comment]
I have tried the latest 5.2.7.53-SNAPSHOT with this tag as well and 
getting the same error.  I have also tried the built in CFIMAP and it 
has issues reading attachment data seemingly sent from exchange 
servers.  A dump fails to show an attachment body- like it is not there.
Any help from anywhere would be greatly appreciated.  I would pay a 
reasonable amount to have this fixed.  Thanks

**Comment by Tropicalista** [comment]
Can you give me more details?

Can I ask what mail provider are you using? Gmail?

I'm testing right now.

**Comment by Tropicalista** [comment]
I have made some fixes. Can you check if this solves your problem?

Thanks

**Comment by HenryGelstor** [comment]
I've tried a few mail servers- I need it to work mostly on zimbra and 
can provide a test account (privately) if need be.

**Comment by HenryGelstor** [comment]
It does!  Thank you!  Works great now. I hope to look at the changes and 
learn something.

There is one more small low impact issue I discovered with weird named 
attachment files on windows only.  I will post a new issue for that.

I have tried the built-in tag in the latest version of Lucee and it has 
flaws that make it unusable as well as different non-standard syntax-- 
making this the best imap implementation for Lucee.  I will post those 
issues w/ Lucee and suggest adopting this as the included tag or a menu 
installed extension.

It's really the only way to complete a project in Lucee requiring IMAP 
right now.

Thank You again Tropicalista!
